node/src/components/consensus/era_supervisor.rs

```diff
@@ -1269,6 +1272,39 @@ where
     }
 }
 
+async fn execute_finalized_block<REv>(
+    effect_builder: EffectBuilder<REv>,
+    finalized_block: FinalizedBlock,
+) where
+    REv: From<StorageRequest> + From<ControlAnnouncement> + From<ContractRuntimeRequest>,
+{
+    // TODO: We already had all these deploys from when we validated the
+    // ProtoBlock. We should just carry them around instead of doing this.
```

Comment:
Are there?  There's like only one unfinalized block at a time, right?  Why do we hold onto them after the proposer's round?

Reply:
Yes, it's unlikely, but it can happen. E.g. if lots of validators are offline, multiple subsequent blocks could be unfinalized at the same time. Or if there's an equivocator there could be several sibling blocks.

We need to hold on to the block for the rest of the era because it will be a dependency of all our future units. If we send someone a unit who is missing a dependency, we need to be able to provide it.

Reply:
Okay, we'll retrieve them from storage later.  I got rid fo this comment in 82de6d9b69d18ed1a4e0467fab6081d5e9089bc2